ic-helpers/src/candid_header.rs

```diff
@@ -0,0 +1,65 @@
+//! This module provides types and functions that help get and verify the structure of a canister
+//! state.
+
+use candid::ser::TypeSerialize;
+use candid::{CandidType, Deserialize};
+use ic_storage::stable::Versioned;
+
+/// Magic prefix used to signify candid encoded binary.
+pub const MAGIC: &[u8] = b"DIDL";
+
+/// Candid header of a versioned state struct.
```

Comment:
What is this struct used for?
How is it used?
I think it might be worth documenting these things

Reply:
This is super nice! Good work